### PR TITLE
XW-3707 | Plista integration

### PR DESCRIPTION
**Pull request by hakubo**

### CODE REVIEW

File: extensions/wikia/Recirculation/js/plista.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		var index = items.findIndex(function (item) {
+			return item.presented_by;
+		});
+
+		return index === -1 && ['AU', 'NZ'].indexOf(Geo.getCountryCode()) > -1;
+	}
+
+	function fetchPlista() {
+		return $.getJSON('http://farm.plista.com/recommendation/?publickey=845c651d11cf72a0f766713f&widgetname=api' +
+			'&count=1&adcount=1&image[width]=583&image[height]=328'
```
Comment (rogatty) [inline extensions/wikia/Recirculation/js/plista.js]: For the other large items in footer recirc we use 320x180. Should we do the same here?

File: extensions/wikia/Recirculation/js/plista.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	}
+
+	function mapPlista(plistaData) {
+		return {
+			meta: 'wikia-impactfooter',
+			source: 'plista',
+			thumbnail: plistaData.img,
+			title: plistaData.title,
+			url: plistaData.url,
+			presented_by: 'Plista',
```
Comment (rogatty) [inline extensions/wikia/Recirculation/js/plista.js:30]: @JamesHawking used `presented_by: item.brand`. Let's unify either way.

File: extensions/wikia/Recirculation/js/plista.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+			isPlista: true
+		}
+	}
+
+	function getPlista() {
+		return fetchPlista().then(mapPlista);
```
Comment (rogatty) [inline extensions/wikia/Recirculation/js/plista.js]: This function seems unnecessary.

File: extensions/wikia/Recirculation/styles/impact-footer.scss [hunk 4]
```diff
@@ -58,9 +58,25 @@ $recirculation-impact-footer-color-feature-overlay: rgba(0, 0, 0, .75);
...
...IE7ocYjYWPvSwQCnCIy6L4rWPcpR1JbZS//XCoXJZgL38scf5Eq4APbGhwlhYWcXV1GqHPaxmo5i6afOWrbZt2yd55Ulgh4Vm+POUcY61vDzsJDlFquV6HXbtrtvq5Ch7UO9RbJHYmzxvdTlymZ6YAGflUw2oM7hR9jRiX+5mKWU/IzmAiqZgJw6yxzPzkw8dPqLmP+2N8gxg5SnTbJR0KnKe1YJUg6NELeITrXjuGNFRO0c5jhPkiRJ/kmidkswWakN7INGpcDzvRunAC+y6HJ1ELyKdldxph7SUaCe6iJu/5I1C3VF8xPrcy4Ck4hU+qsV8dj0DPza/B5v4oxKOchYJXihn+S0DMUS90g6uvwqzC6UPASI0LeeIKtDH9ddQydPwPHNtVNV4ZmooBlNBEwmny/Q6a3m6fq4fFp6kaFwgspWKORs5ShfCvVBExAcwSFG4qD3Lh1VDKUaw2+4UYhdyKcEllipGNA5diJmwjMsrGyToUzWULIaiFnA1BS5hX7b7CfM3OMp++LlL/vKb8lgFDWCYYuDQXvQiTkPJw1vubw2618NeNnp7aAn7554+b5RZADEUWS9nMhAo7xD/n4udQuwsXk5JNNZ3RQWN4D3X9Lmo8UBZg0btxRVFr6G0urxymSUBijy8KS8+b57gYBfKbo7CingFJEViWwGPxiK8r9lks5Q7gg+4+qFAKUO01veOmVHmgUKrcIbrV3kMBMowMvgD/RWbTCmSKOnaa61E15N4QkTUwzIZxWChvCDecg8ZxC7dHRQbqNl8hP7TzQgwEW0v17qtA2mD7D7cgRGLJ0obOlilmfah2l1BObjBrh8jm2gt2tlLSxovr1GwPKYqo20qrzSHgY1yTK6uEoptbiP+70lwNn1SXqTtQd2fQro0MKLAzLqFBwprHZ91ZtowOtZXrtaj/fN07GK7lfUEtmR+UYQDFt5XdF+23PSTzk2BUsHbm238RNeL426jrxDP0sJhb7jW7NqZLV9/U1ga9u11K3mhiIaan3NjQret+s6mNNwUtWJp+96e7KgWs3748Sd1zkDXeOqMp61OkUp/9uWhheWGiaH8AlXCrx5b4T+PAoxmRf5m7cfvqXrfHnAZigTFyP3h65M2P9uq6CE59c9t/2ap4F5Z72sKIcUIGrLofXzW/1+mL38A8Q75z/jUvY4AAAAASUVORK5CYII=');
```
Comment (rogatty) [inline extensions/wikia/Recirculation/styles/impact-footer.scss]: Can we use `` instead?

File: extensions/wikia/Recirculation/templates/client/impactFooter.mustache [hunk 5]
```diff
@@ -1,75 +1,87 @@
...
+							Sponsored
+						</div>
+					{{/presented_by}}
+					<div class="thumbnail placeholder-thumbnail">
+						{{#thumbnail}}
+							<img src="{{thumbnail}}" alt="{{title}}">
+						{{/thumbnail}}
+					</div>
+					{{#isPlista}}
+						<a href="https://www.plista.com/optout/index" class="badge badge-plista"></a>
```
Comment (rogatty) [inline extensions/wikia/Recirculation/templates/client/impactFooter.mustache]: https://www.plista.com/optout is an actual, working page.

File: extensions/wikia/Recirculation/templates/client/impactFooter.mustache [hunk 6]
```diff
@@ -1,75 +1,87 @@
...
-      </ul>
+			<ul class="discussion-threads">
+				{{#posts}}
+					<li class="discussion-thread" data-index="{{index}}" data-source="{{source}}"><a
+						class="track-discussions" href="{{url}}">
+						<div class="discussion-author">
+							<img src="{{meta.authorAvatarUrl}}" class="discussion-author-avatar"/>
+							<h4>{{author}}</h4>
+							<span class="subtext">&#183; <time class="discussion-timestamp"
+															   datetime="{{pub_date}}"></time></span>
```
Comment (rogatty) [inline extensions/wikia/Recirculation/templates/client/impactFooter.mustache]: This is a very strange line break.

File: extensions/wikia/Recirculation/js/plista.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
+define('ext.wikia.recirculation.plista', ['jquery'], function ($) {
+	function shouldFetchPlista(items) {
+		var index = items.findIndex(function (item) {
```
Comment (WikiaTech) [inline extensions/wikia/Recirculation/js/plista.js]: MAJOR Remove the declaration of the unused 'index' variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedVariable)
MAJOR Remove this useless assignment to local variable "index" [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)

File: extensions/wikia/Recirculation/js/plista.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+			return item.presented_by;
+		});
+
+		return true;//index === -1 && ['AU', 'NZ'].indexOf(Geo.getCountryCode()) > -1;
```
Comment (WikiaTech) [inline extensions/wikia/Recirculation/js/plista.js]: MAJOR Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: extensions/wikia/Recirculation/js/plista.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+					"brand": "Life Insurance Comparison",
+					"img": "http://static.plista.com/image/resized/244310/a9q1eFGqTxGvhcD_356x200_8365.jpg (32kB)",
+					"status": 34,
+					"text": "New Life Insurance Comparison Site Helps Aussies Save Big Money Quickly!",
+					"title": "Over 40 with Life Insurance? You need to read this",
+					"type": "pet",
+					"url": "http://click.plista.com/pets/?friendid=0&frienddomainid=243030&widgetid=45044&itemid=371828933&campaignid=257204&bucketid=0&rh=59770fddc9cc34.78975602&lh=59770fde620b32.28763435&bv=_0_bVMLruQgDLvOqxShOB9C5mx7-DW0o5ldPVq1BdvBIamIoBPgQ7DUsWxJeLqZWCLgLXp5RCbE5moVDohzQJAL0A6BTnWTaN06ElK1K3W2MIxNXSRDVLyj0BPkWLg6w6GcUhEdyPCJ_qMjls3wMZRXnifG-Lkw0K1YnPBztI5uvpUYLhsggprjLOWY-mDX6y0sxkZ7qT8s908IhjedvPMXjNGXV7Y9WHxhL7sN25dzbAp0J6X5EP16-U2I7ePj1u1_alwvymtVQjd3Ganr3hvnehyzYtl1A5EfxA6CeA7jS-PDzAr2zuSDyKntCmuWRjA1e9WUSI0gErtX1CacE67RFKvIiV2WbJ-jhi_Tkl1Xq8w5qXPz5qwslHIPk516IPZmXlECPyGouUN1kcblncFcchtr1nAvdobV2hZuuxrMRwwKD3YYlDFto7Y3DlNf7KoKbTmK87JVOnd77s777ldqqhryDLAhNHTtr52R7sxm5FLDP31uitLiP9N00HdY3nqOoqtJ9LN6kanFRv8L&tend=1501003870&crc=b64bbdc21f80c3b8860cd862d9980088"
+				};
+
+				if (data[0]) {
```
Comment (WikiaTech) [inline extensions/wikia/Recirculation/js/plista.js]: MAJOR Remove this code after the "return" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnreachableCode)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 4 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**